static/app/views/automations/components/actionFilters/taggedEvent.tsx

```diff
@@ -33,16 +39,54 @@ function KeyField() {
   const {condition, condition_id, onUpdate} = useDataConditionNodeContext();
   const {removeError} = useAutomationBuilderErrorContext();
 
+  // TODO - get form context, get the detector ids from the context.
+  // if there are connected detectors / projects, then grab the project id list and use that. otherwise, -1.
+  const projectIds = ['-1'];
+
+  // Select all the tags for an organization to generate a list of the most likely tags
+  const organization = useOrganization();
+  const {data: tagOptions = [], isLoading} = useFetchOrganizationTags(
+    {
+      orgSlug: organization.slug,
+      projectIds,
+      dataset: Dataset.ISSUE_PLATFORM,
+      useCache: true,
+      keepPreviousData: true,
+    },
+    {}
+  );
+
+  const sortedTags = useMemo(() => {
+    const sorted = tagOptions.toSorted((a, b) => {
+      return (a.totalValues || 0) > (b.totalValues || 0) ? -1 : 1;
+    });
+
+    if (
+      condition.comparison.key &&
+      !sorted.some(tag => tag.key === condition.comparison.key)
+    ) {
+      sorted.unshift(condition.comparison);
+    }
+
+    return sorted;
+  }, [tagOptions, condition.comparison]);
+
   return (
-    <AutomationBuilderInput
+    <AutomationBuilderSelect
+      disabled={isLoading}
+      creatable
       name={`${condition_id}.comparison.key`}
-      placeholder={t('tag')}
-      value={condition.comparison.key ?? ''}
-      onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
-        onUpdate({comparison: {...condition.comparison, key: e.target.value}});
+      aria-label={t('Tag')}
+      placeholder={isLoading ? t('Loading tags...') : t('tag')}
```

Comment:
we use the ... symbol in these
```suggestion
      placeholder={isLoading ? t('Loading tags\u2026') : t('tag')}
```